model/src/main/kotlin/io/spine/validation/bound/RangeOption.kt

```diff
@@ -109,18 +109,17 @@ internal class RangePolicy : Policy<FieldOptionDiscovered>() {
         val fieldType = checkFieldType(field, file, RANGE)
 
         val option = event.option.unpack<RangeOption>()
-        val (messageType, _) = typeSystem.findMessage(field.declaringType)!!
-        val context = RangeContext(option.value, typeSystem, messageType, fieldType, field, file)
-        val delimiter = context.checkDelimiter()
+        val optionMetadata = RangeOptionMetadata(option.value, field, fieldType, file, typeSystem)
+        val delimiter = optionMetadata.checkDelimiter()
 
-        val (left, right) = context.range.split(delimiter)
-        val (lowerExclusive, upperExclusive) = context.checkBoundTypes(left, right)
-        val lower = context.checkNumericBound(left.substring(1), lowerExclusive)
-        val upper = context.checkNumericBound(right.dropLast(1), upperExclusive)
+        val (left, right) = optionMetadata.range.split(delimiter)
+        val (lowerExclusive, upperExclusive) = optionMetadata.checkBrackets(left, right)
+        val lower = optionMetadata.checkNumericBound(left.substring(1), lowerExclusive)
```

Comment:
What are those `left.substring(1)` and `right.dropLast(1)` are about?

Reply:
So that to avoid passing bound values with inclusive/exclusive brackets `][)(`.

Passing with brackets would have forced the parser to check for `([` at the beginning and `)]` at the end. 

Min/max options also use this method, they don't have such symbols at all.

So, the `(range)` option has its own `optionMetadata.checkBrackets(left, right)` compilation check.